app/Cart.php

```diff
@@ -2,35 +2,58 @@
 
 namespace App;
 
+use Illuminate\Support\Facades\Session;
+
 
 class Cart
 {
-    public $items = null;
-    public $totalQty = 0;
+    public $items = [];
+    public $totalQuantity = 0;
     public $totalPrice = 0;
 
-    public function __construct($oldCart)
-    {
-    	if($oldCart) {
-			$this->items = $oldCart->items;
-    		$this->totalQty = $oldCart->totalQty;
-    		$this->totalPrice = $oldCart->totalPrice;
-    	}
+    public function __construct(){
+        $this->load();
+    }
+
+    public function get(){
+        return $this->load();
+    }
+
+    private function load(){
+        $data = Session::has('cart') ? Session::get('cart') : new \stdClass;
+        if(!$data) return;
```

Comment:
Bro this condition will fail because, look at this 
$data = Session::has('cart') ? Session::get('cart') : new \stdClass;
So if there is no session with cart index the statement will return stdClass and this will make the $data variable always true the the condition will fail and this will lead to Bug $data::items property undefined, so I've modified this condition to $data = Session::has('cart') ? Session::get('cart') : false; and then if the yes (!$data) I will instantiate new stdCalss
with the needed property and assign that property to the corresponding class property like so
$data->items = $this->items; and so on